src/qcodes/dataset/database_extract_runs.py

```diff
@@ -168,3 +175,218 @@ def _extract_single_dataset_into_db(
+                
+                # Try to export dataset to NetCDF and create metadata-only version
+                status = _process_single_dataset(
+                    dataset, source_conn, target_conn, export_path, target_exp_id
+                )
+                result_status[run_id] = status
+                
+            except Exception as e:
+                log.exception(f"Failed to process dataset {run_id}")
+                result_status[run_id] = f"failed: {str(e)}"
+        
+        log.info(f"Processing complete. Status summary: {result_status}")
+        return result_status
+        
+    except Exception as e:
+        log.exception(f"Database operation failed")
+        raise
+    finally:
+        if source_conn is not None:
+            source_conn.close()
+        if target_conn is not None:
+            target_conn.close()
+
+
+def _process_single_dataset(
+    dataset: DataSet,
+    source_conn: AtomicConnection,
+    target_conn: AtomicConnection,
+    export_path: Path,
+    target_exp_id: int,
+) -> str:
+    """
+    Process a single dataset: export to NetCDF and create metadata-only version
+    or copy as-is if export fails.
+    
+    Returns:
+        Status string indicating what was done with the dataset
+    """
+    run_id = dataset.run_id
+    
+    # Check if dataset is already in target database
+    existing_run_id = get_runid_from_guid(target_conn, dataset.guid)
+    if existing_run_id is not None:
+        log.info(f"Dataset {run_id} (GUID: {dataset.guid}) already exists in target database")
+        return "already_exists"
+    
+    # Check if dataset is completed
+    if not dataset.completed:
+        log.warning(f"Dataset {run_id} is not completed, copying as-is")
+        return _copy_dataset_as_is(dataset, target_conn, target_exp_id)
+    
+    try:
+        # Check if dataset has already been exported to NetCDF
+        existing_netcdf_path = dataset.export_info.export_paths.get("nc")
+        
+        if existing_netcdf_path is not None:
+            # Check if the existing export path matches the desired export path
+            existing_path = Path(existing_netcdf_path)
+            
+            # If the file exists at the expected location, skip export
+            if existing_path.exists() and existing_path.parent == export_path:
+                log.info(f"Dataset {run_id} already exported to NetCDF at {existing_netcdf_path}")
+                netcdf_export_path = existing_netcdf_path
+            else:
+                # Export to the new location
+                log.info(f"Dataset {run_id} was exported to different location, re-exporting to {export_path}")
+                dataset.export("netcdf", path=export_path)
+                netcdf_export_path = dataset.export_info.export_paths.get("nc")
+        else:
+            # Try to export to NetCDF
+            log.info(f"Attempting to export dataset {run_id} to NetCDF")
+            dataset.export("netcdf", path=export_path)
```

Comment:
Could you refactor this function so that the dataset.export call is not duplicated? I think it's easier to read the code that way